pkgs/development/compilers/llvm/16/libcxxabi/default.nix

```diff
@@ -67,6 +67,7 @@ stdenv.mkDerivation rec {
     "-DCMAKE_CXX_COMPILER_WORKS=ON"
   ] ++ lib.optionals (stdenv.hostPlatform.useLLVM or false) [
     "-DLLVM_ENABLE_LIBCXX=ON"
+  ] ++ lib.optionals (!stdenv.isDarwin) [
```

Comment:
Darwin isn’t `useLLVM`, so this can continue to be used unconditionally.

Reply:
I removed this per your suggestion. But doing so breaks `pkgs.libcxxabi` (per failing checks marked on the review) Did we both misread this?

Misread:

"The addition is not necessary becuase stdenv.hostPlatform.useLLVM is mutually exclusive with stdenv.isDarwin"

Actual effect / intent:

"All non-darwin platforms gain `-DLIBCXXABI_USE_LLVM_UNWINDER=ON`".

I've reinstated this in cdc598cc33acfcd8ec2de3d35307fe845c62bccc.

(Otherwise they fail with various undefined references to `_Unwind_*` functions: (full list: `_Unwind_DeleteException _Unwind_GetIP _Unwind_GetLanguageSpecificData _Unwind_GetRegionStart _Unwind_RaiseException _Unwind_Resume _Unwind_SetGR _Unwind_SetIP`)

Reply:
I don't think I'm at the bottom of this. I'm seeing something *really* weird. nix-diff shows the only difference between these two drvs is:

```diff

- /nix/store/ihnyk8aph48r3n25sp62iwh3bd91qpgr-libcxxabi-16.0.6.drv:{out}
+ /nix/store/hpl0mrz00kq3pixj0f1n7n41a82j0fh9-libcxxabi-16.0.6.drv:{out}
• The input derivation named `libunwind-16.0.6` differs
  - /nix/store/6w3yyn5fpbnz30v522a2c28ab3dk1yil-libunwind-16.0.6.drv:{dev}
  + /nix/store/6n4mvbg01qwmjn4nlnhqaihg2qlil91m-libunwind-16.0.6.drv:{dev}
  • The environments do not match:
      + postInstall=# libcxxabi wants to link to libunwind_shared.so (?).
ln -s $out/lib/libunwind.so $out/lib/libunwind_shared.so

• Skipping environment comparison

```


But then if I try to build the one with the extra postInstall, I get link errors related to unwind symbols. This makes no sense: these errors are happening before postInstall, and the drv are otherwise identical.

Reply:
Oh, hmm, it seems nix-diff doesn't show environment comparisons by default. I've added `-DCMAKE_EXE_LINKER_FLAGS=-nostdlib -DCMAKE_SHARED_LINKER_FLAGS=-nostdlib`; previously compiler-rt cmake was supplying `-lgcc_s` in order to acquire the unwind symbols, but that's not happening if `-nostdlib` is passed, so I need to change the condition to `((stdenv.hostPlatform.useLLVM or false) || stdenv.isDarwin)`, or leave the darwin bootstrap alone and put the condition in with the other `(stdenv.hostPlatform.useLLVM or false)` parts.

Reply:
d72b64dedd6a056261f0dfa64c4ecd7b96cab85a fixes this, with my first suggestion. I've left the `-nostdlib` in the definition of libcxxabi since it feels more like a libcxxabi issue which could conceivably be fixed some day.